src/btree/bt_random.c

```diff
@@ -302,11 +305,15 @@ __wt_btcur_next_random(WT_CURSOR_BTREE *cbt)
 	WT_UPDATE *upd;
 	wt_off_t size;
 	uint64_t n, skip;
+	uint32_t descent_flags;
 	bool valid;
 
 	btree = cbt->btree;
 	cursor = &cbt->iface;
 	session = (WT_SESSION_IMPL *)cbt->iface.session;
+	descent_flags = 0;
+	if (F_ISSET(cbt, WT_CBT_READ_WONT_NEED))
+		FLD_SET(descent_flags, WT_CBT_READ_WONT_NEED);
```

Comment:
If you take my suggestion this would become `WT_DESCENT_READ_ONCE` (or maybe call it `WT_DESCENT_READ_WONT_NEED`).